src/components/storage-location.js

```diff
@@ -12,15 +14,35 @@ import { getParentPathForStorageLocationPath, getStorageLocationPathForUrl } fro
 
 function StorageLocation({ storageLocation, initialUrl, isFullscreen, onEnterFullscreen, onExitFullscreen, onSelect, onCancel }) {
   const { t } = useTranslation('storageLocation');
+  const setStorageLocation = useSetStorageLocation();
   const storageApiClient = useSessionAwareApiClient(StorageApiClient);
 
   const [files, setFiles] = useState([]);
-  const [selectedFile, setSelectedFile] = useState(null);
   const [isLoading, setIsLoading] = useState(false);
+  const [selectedFile, setSelectedFile] = useState(null);
   const [parentDirectory, setParentDirectory] = useState(null);
   const [currentDirectory, setCurrentDirectory] = useState(null);
   const [currentLoadedDirectoryPath, setCurrentLoadedDirectoryPath] = useState(null);
-  const [filesViewerDisplay, setFilesViewerDisplay] = useState(FILE_VIEWER_DISPLAY.list);
+  const [filesViewerDisplay, setFilesViewerDisplay] = useState(FILE_VIEWER_DISPLAY.grid);
+
+  const fetchStorageContent = useCallback(async () => {
+    if (!currentLoadedDirectoryPath) {
+      return;
+    }
+
+    try {
+      let result = await storageApiClient.getCdnObjects(currentLoadedDirectoryPath);
+      if (!result.objects.length) {
```

Comment:
I think I don't completely understand the purpose of this fallback. It seems it makes opening an empty folder impossible? Or did I get it wrong?

Reply:
We actually never have empty "folders", this doesn't exist on the CDN.
This check is for the case when one types in a wrong path, so we fall back onto the initial storage location path.
For example:
`cdn://media/documentThatDoesNotExist"`

Reply:
As discussed, we have to find a way how to show an actually valid empty folder (e.g. immediately after creating a new document). For now let's leave it this way until we have figured out a way to implement this.